model_compression_toolkit/core/common/quantization/set_node_quantization_config.py

```diff
@@ -329,36 +337,137 @@ def filter_qc_options_with_manual_bit_width(
+            # If a base_config with the specified activation_manual_bit_width_override exists in the node_qc_options_list,
             # point the base_config to this option.
             base_config = node_qc_options_list[node_qc_options_list.index(updated_base_config)]
         else:
             # Choose a different configuration from node_qc_options_list. If multiple options exist, issue a warning.
             base_config = node_qc_options_list[0]
             if len(node_qc_options_list) > 0 and not mixed_precision_enable:
                 Logger.info(
-                    f"Request received to select {manual_bit_width_override} activation bits. However, the base configuration for layer type {node.type} is missing in the node_qc_options_list."
-                    f" Overriding base_config with an option that uses {manual_bit_width_override} bit activations.")  # pragma: no cover
-    return base_config, node_qc_options_list
\ No newline at end of file
+                    f"Request received to select {activation_manual_bit_width_override} activation bits. However, the base configuration for layer type {node.type} is missing in the node_qc_options_list."
+                    f" Overriding base_config with an option that uses {activation_manual_bit_width_override} bit activations.")  # pragma: no cover
+
+    return base_config, node_qc_options_list
+
+
+def filter_weights_qc_options_with_manual_bit_width(
+        node: BaseNode,
+        node_qc_options_list: List[OpQuantizationConfig],
+        base_config: OpQuantizationConfig,
+        weights_manual_bit_width_override: Optional[list],
+        mixed_precision_enable: bool) -> Tuple[OpQuantizationConfig, List[OpQuantizationConfig]]:
+    """
+    Update the weights quantization configurations for a node, allowing manual bit-width overrides if specified.
+
+    Args:
+        node (BaseNode): A node to set quantization configuration candidates to.
+        node_qc_options_list (List[OpQuantizationConfig]): List of quantization configs for the node.
+        base_config (OpQuantizationConfig): Base quantization config for the node.
+        weights_manual_bit_width_override (Optional[Dict]): Specifies a custom bit-width to override the node's weights bit-width.
+        mixed_precision_enable (bool): Whether mixed precision is enabled.
+
+    Returns:
+        Tuple[OpQuantizationConfig, List[OpQuantizationConfig]]: The updated base configuration and the filtered list of quantization configs.
+    """
+    if not weights_manual_bit_width_override:
+        return base_config, node_qc_options_list
+
+    # Filter node_qc_options_list to retain only the options with weights bits equal to weights_manual_bit_width_override.
+    node_qc_options_weights_list = []
+    override_attr, override_bitwidth = [], []
+
+    for weights_manual_bit_width in weights_manual_bit_width_override:
```

Comment:
If I followed the code correctly, `weights_manual_bit_width_override` is a list with two items - [bit_width, attribute_name].
So, first of all - modify the function description and typehint.
Second, we do you need to decompose the pairs into two lists and then zip the pairs together for the iteration? seems like you can just iterate over the original list